[Dioclei] iP

Dioclei opened the pull request:
# Fluke
* Creates tasks for you, this includes:
> Albeit with a _little_ [randomness](https://random.org)
1. Todo
2. Deadline (has a ending date)
3. Event (has a starting date and ending date)

`test code`
`int i = 2`

Main function: 😄 👍 
```java
public static void main(String[] args) {
        // initialise Fluke
        Fluke fluke = new Fluke(SAVE_FILE_NAME);
        fluke.run();
    }
```

Some possible **future** ~~plans~~ plans:
- [x] implement OOP stuff
- [ ] Add Command class for better OOP

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Dioclei committed:
Renamed Duke to Fluke. Added skeletal code.

Dioclei committed:
Give Fluke a personality

Dioclei committed:
Add ability to give Fluke commands

Dioclei committed:
input now adds to a list.

Dioclei committed:
added tasks and renamed methods appropriately.

Dioclei committed:
Updated with 3 types of tasks

Dioclei committed:
added automated testing library

Dioclei committed:
added delete

Dioclei committed:
add enum for commands

Dioclei committed:
reformatted code

Dioclei committed:
add error handling for invalid inputs for event and deadline

Dioclei committed:
Save todo list data to fluke.txt and read data on startup

Dioclei committed:
change storage of date data to use LocalDate

Dioclei committed:
refactor code, add parser storage tasklist and ui. rename duke to fluke

Dioclei committed:
rename DukeException to FlukeException

Dioclei committed:
fix ui bug with printing list

Dioclei committed:
Merge branch 'MoreOOP'

* MoreOOP:
  fix ui bug with printing list
  rename DukeException to FlukeException
  refactor code, add parser storage tasklist and ui. rename duke to fluke

Dioclei committed:
cleanup code

Dioclei committed:
refactor into packages

Dioclei committed:
Merge branch 'Packages'

* Packages:
  refactor into packages
  cleanup code

Dioclei committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

* commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552':
  Bump gradle and lib version
  Add Gradle support

Dioclei committed:
remove redundant constructor for fluke

Dioclei committed:
fix mark as done bug

Dioclei committed:
add tests for TaskList

Dioclei committed:
add tests for Parser

Dioclei committed:
Merge branch 'branch-testing'

* branch-testing:
  add tests for Parser
  add tests for TaskList

Dioclei committed:
Add javadoc to all public methods and a few non-trivial private methods.

Dioclei committed:
comply with coding standards

Dioclei committed:
implement find feature

Dioclei committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add javadoc to all public methods and a few non-trivial private methods.

Dioclei committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  comply with coding standards

# Conflicts:
#	src/main/java/fluke/Fluke.java
#	src/main/java/fluke/Parser.java
#	src/main/java/fluke/TaskList.java
#	src/main/java/fluke/Ui.java
#	src/main/java/fluke/tasks/Deadline.java
#	src/main/java/fluke/tasks/Event.java

Dioclei committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  implement find feature

# Conflicts:
#	src/main/java/fluke/TaskList.java
#	src/main/java/fluke/Ui.java

Dioclei committed:
add checkstyle gradle script

Dioclei committed:
Merge tag 'A-CheckStyle'

* tag 'A-CheckStyle':
  add checkstyle gradle script

Kokseng1 commented on src/main/java/fluke/Ui.java:
```diff
@@ -0,0 +1,125 @@
+/**
+ * This class contains all user interface related functionalities.
+ */
+public class Ui {
```
Good practice adding all the standard command inputs into the UI

Kokseng1 commented on src/main/java/fluke/exceptions/FlukeException.java:
```diff
@@ -0,0 +1,14 @@
+package fluke.exceptions;
```
Nice job creating different exceptions for specific errors

Kokseng1 commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,4 @@
+todo do homework
+event house event /from tomorrow /to friday
+deadline prepare for test /by friday night
+bye
```
nitpicking at this point, but maybe you can consider adding more tests, eg. marking or deleting tasks

Kokseng1 commented on src/main/java/fluke/Parser.java:
```diff
@@ -0,0 +1,272 @@
+     * @throws InvalidInputException if the command given is invalid.
+     */
+    public static Fluke.Command parseCommand(String command) throws InvalidInputException {
+        if (command.equals("bye")) {
```
not wrong, but why not use case and switch?

Kokseng1 reviewed (commented):
No errors found, code is easily readable and follows coding standard and quality. Goodjob!

flexibo commented on src/main/java/fluke/Fluke.java:
```diff
@@ -0,0 +1,212 @@
+    }
+
+    /**
+     * Helper function for additional logic related to changing the list.
```
Nice use of commenting to describe each method and what it does. Perhaps add a new line between the description and the params to fit the coding standard.

flexibo commented on src/main/java/fluke/Fluke.java:
```diff
@@ -0,0 +1,212 @@
+    }
+
+    /**
+     * Fluke attempts to find tasks with the corresponding keyword.
```
Nice use of comments to describe what each method does, maybe possible to add a line between the description and the params to fit the module's code style. 
```suggestion
     * Fluke attempts to find tasks with the corresponding keyword.
     
```
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/40513197/cd7b77e4-b3bb-4eec-a649-41981f23001c)

flexibo commented on src/main/java/fluke/Parser.java:
```diff
@@ -0,0 +1,272 @@
+     * @throws FlukeException if the input is invalid.
+     */
+    public static int parseDeleteCommand(String command) throws FlukeException {
+        if (command.length() <= 7) {
```
Is it easier to make use of split function in Java so you can count number of words instead of characters instead and make the comparison there? 

[Here is a link for more info about split strings](https://www.geeksforgeeks.org/split-string-java-examples/)

so for example 
```java 
String[] splitText = command.split(" "); // Split command string with spaces
if (splitText.length > 1) { 
   // ...
} 
```

flexibo commented on src/main/java/fluke/Parser.java:
```diff
@@ -0,0 +1,272 @@
+import fluke.tasks.Deadline;
+import fluke.tasks.Event;
+import fluke.tasks.Task;
+import fluke.tasks.Todo;
```
Nice job in making sure that the import statements are in order.

flexibo commented on src/main/java/fluke/Parser.java:
```diff
@@ -0,0 +1,272 @@
+    }
+
+    /**
+     * Parses and validates a Deadline command
```
Nice job using "parses" and "validates" instead of "parse" or "parsing" 😄

flexibo commented on src/main/java/fluke/TaskList.java:
```diff
@@ -0,0 +1,181 @@
+ * various methods related to managing the list of tasks, such as adding and deleting.
+ */
+public class TaskList {
+    private final ArrayList<Task> listOfTasks;
```
while ``listOfTasks`` is not wrong, might be better to rename to ``tasks`` instead due to this rule for coding standard.
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/40513197/e4dc686a-929f-4ad4-8664-4169a350bf0f)

flexibo reviewed (commented):
Overall good code, just a few nitpicks regards to coding standard for improvement.

Dioclei committed:
Add GUI to Fluke

Dioclei committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI to Fluke

Dioclei committed:
Change build.gradle config

Dioclei committed:
Add assertions for assumptions in the code

There are no assertions in the code to make sure of key assumptions.
This can cause errors which are hard to debug such as NullPointerExceptions.

Add assertions to make sure that:
1. fluke and ui elements are not null upon initialisation and upon user input.
2. save file parser regex correctly detects what is needed, e.g. the type of task.

Reasons for the above:
1. This is important because the whole application will break when fluke and ui elements are null.
2. This is important because the regex needs to work to parse the whole save file correctly.

Dioclei committed:
Refactor code to make it more readable

Dioclei committed:
Revert "Refactor code to make it more readable"

This reverts commit 97c24ce0dece2daf7fd1a6467815a0a20302afbc.

Dioclei committed:
Revert "Revert "Refactor code to make it more readable""

This reverts commit 81f7576f584cf7be3bb56f0418dc8acaae5adca3.

Dioclei committed:
Merge pull request #1 from Dioclei/branch-A-Assertions

Add assertions for assumptions in the code

Dioclei committed:
Merge branch 'master' of https://github.com/Dioclei/ip

* 'master' of https://github.com/Dioclei/ip:
  Add assertions for assumptions in the code

Dioclei committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions for assumptions in the code

Dioclei committed:
Merge pull request #2 from Dioclei/branch-A-CodeQuality

Improve code quality by refactoring some code and making it more readable

Dioclei committed:
Merge branch 'master' of https://github.com/Dioclei/ip

* 'master' of https://github.com/Dioclei/ip:
  Revert "Revert "Refactor code to make it more readable""
  Revert "Refactor code to make it more readable"
  Refactor code to make it more readable

Dioclei committed:
Add yml file for github actions

We don't have automatic continuous integration.
This will help to make our app code more robust.

Dioclei committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Add yml file for github actions

Dioclei committed:
Add mass operations for simple operations

Sometimes users may want to do simple operations multiple times.
This includes:
* Mark task as done
* Mark task as not done
* Delete task
This change allows users to input numbers separated by commas for the above 3 operations.
For instance, if the user wants to delete tasks 1, 3, and 5, they can simply write "delete 1, 3, 5"

Dioclei committed:
Merge branch 'branch-C-MassOps'

* branch-C-MassOps:
  Add mass operations for simple operations

Dioclei committed:
Add user guide

Users may not know how to use Fluke.
The user guide aims to show the user how to use Fluke, by documenting important commands.

Dioclei committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add user guide
